### ISSUE TITLE
Increase dependency versions where not covered by auto-update

### ISSUE THREAD

**Issue by kate-goldenring** [deislabs/akri#324]
Akri's dependencies should be updated. While our auto-update workflow follows semver for updating patches and minor versions where possible, the following crate versions need to be updated:
- [ ] udev to 0.6
- [x] async-stream in Agent
- [x] tonic, tokio, warp (0.3 requires tokio 1.0), futures, prost, hyper which are [interdependent and non-trivial to bump](https://github.com/deislabs/akri/issues/223)
- [x] actix in Webhook 
- [x] yaserde in ONVIF discovery handler

Akri's Python and .NET sample brokers and apps should also be inspected for version updates.

**Comment by kate-goldenring** [comment]
In the udev Discovery Handler, a `Enumerator` trait was created in order to mock the `udev` crate's [`Enumerator` struct](https://docs.rs/udev/0.6.2/udev/struct.Enumerator.html#method.scan_devices). Updating `udev` to 0.6.0 requires changing the `scan_devices` [function definition in the Trait](https://sourcegraph.com/github.com/deislabs/akri/-/blob/discovery-handlers/udev/src/wrappers.rs#L101) to match the [new definition](https://docs.rs/udev/0.6.2/udev/struct.Enumerator.html#method.scan_devices).
FROM
`fn scan_devices(&mut self) -> std::io::Result<udev::Devices>;`
TO
`fn scan_devices(&mut self) -> Result<List<'_, Enumerator, Device>>;`
which requires reconciling lifetimes

**Comment by kate-goldenring** [comment]
We cannot update to the latest versions of kube-rs and k8s-openapi because `kube-rs` requires that all fields of a CRD Spec derive `JsonSchema`; however, `k8s-openapi` structs do not. Consequently, any CRD specs that contain K8s objects (such as `PodSpec` and `ServiceSpec` structs in ours) do no compile. There is an issue up on k8s-openapi to support `JsonSchema` that will need to be resolved before we can proceed with updating our dependencies.  This fix https://github.com/Arnavion/k8s-openapi/pull/104 works as I am able to compile everything by patching `k8s-openapi` with the PR branch.

**Comment by kate-goldenring** [comment]
Closing due to all dependencies being updated besides `udev` which now has its own issue: https://github.com/deislabs/akri/issues/364
